src/journal-remote/journal-gatewayd.c

```diff
@@ -885,6 +885,66 @@ static int request_handler_machine(
         return MHD_queue_response(connection, MHD_HTTP_OK, response);
 }
 
+static int request_handler_boots(
+                struct MHD_Connection *connection,
+                void *connection_cls) {
+
+        _cleanup_(MHD_destroy_responsep) struct MHD_Response *response = NULL;
+        RequestMeta *m = ASSERT_PTR(connection_cls);
+        int r;
+        _cleanup_free_ LogId *boots = NULL;
+        size_t n_boots;
+        _cleanup_(sd_json_variant_unrefp) sd_json_variant *array = NULL;
+        _cleanup_free_ char *json = NULL;
+
+        assert(connection);
+
+        r = open_journal(m);
+        if (r < 0)
+                return mhd_respondf(connection, r, MHD_HTTP_INTERNAL_SERVER_ERROR, "Failed to open journal: %m");
+
+        r = journal_get_boots(
+                        m->journal,
+                        false,
+                        SIZE_MAX,
+                        &boots, &n_boots);
+
+        if (r < 0)
+                return mhd_respondf(connection, r, MHD_HTTP_INTERNAL_SERVER_ERROR, "Failed to determine boots: %m");
+        if (r == 0)
+                return mhd_respondf(connection, r, MHD_HTTP_INTERNAL_SERVER_ERROR, "No boot found.");
```

Comment:
this should result in an empty array, no?

Reply:
It's considered an error (`ENODATA`) in `journalctl` too: https://github.com/systemd/systemd/blob/eea9eb1f050b638229f78c7d1f1cd09c2422693b/src/journal/journalctl-misc.c#L181

I'm actually not sure how it can happen there won't be any boots but if you think we should be less strict here and just return nothing (maybe along with a logging entry?) I'm fine with that.